openvdb/points/PointDataGrid.h

```diff
@@ -1163,73 +1163,103 @@ PointDataLeafNode<T, Log2Dim>::readBuffers(std::istream& is, bool fromHalf)
     this->readBuffers(is, CoordBBox::inf(), fromHalf);
 }
 
+
+////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////
+
+
+namespace page_stream
+{
+
+template <typename StreamT>
+inline void destroyPagedStream(const MetaMap& auxData, const Index index)
+{
+    // if paged stream exists, delete it
+    const std::string key("paged:" + std::to_string(index));
+    (const_cast<MetaMap&>(auxData)).removeMeta(key);
+}
+
+template <>
+inline void destroyPagedStream<compression::PagedOutputStream>
+    (const MetaMap& auxData, const Index index)
+{
+    using TypedMetaT = TypedMetadata<compression::PagedOutputStream::Ptr>;
+
+    // if paged stream exists, flush and delete it
+    const std::string key("paged:" + std::to_string(index));
+    const TypedMetaT::ConstPtr it = auxData.getMetadata<TypedMetaT>(key);
+    if (it) {
+        it->value()->flush();
+        (const_cast<MetaMap&>(auxData)).removeMeta(key);
+    }
+}
+
+template <typename StreamT>
+inline StreamT&
+getOrInsertPagedStream(const MetaMap& auxData,
+                       const Index index)
+{
+    using TypedMetaT = TypedMetadata<typename StreamT::Ptr>;
+
+    const std::string key("paged:" + std::to_string(index));
+    typename TypedMetaT::ConstPtr it = auxData.getMetadata<TypedMetaT>(key);
```

Comment:
note - `getMetadata` performs string type checks. If we don't want to do that, we can use `static_cast<...>(auxData[key])`